### PR TITLE
Post-processing improvements for asynchronous environment

### PR DESCRIPTION
**Pull request by knopers8**
Not to be merged yet, I am just taking advantage of CI.

### CODE REVIEW

**Comment by knopers8** [comment]
These are not all improvements that are upcoming, but we could already start with those. ready for review.

File: Framework/include/QualityControl/QualityObject.h [hunk 1]
```diff
@@ -124,7 +124,7 @@ class QualityObject : public TObject
   void setQuality(const Quality& quality);
   const std::string& getCheckName() const;
   const std::string& getPolicyName() const;
-  const std::vector<std::string> getMonitorObjectsNames() const;
+  const std::vector<std::string>& getMonitorObjectsNames() const;
```
Comment (Barthelemy) [inline Framework/include/QualityControl/QualityObject.h:127]: should the class version be increased ?

**Review by Barthelemy — APPROVED** [review]
Thank you this is very nice.

File: Framework/include/QualityControl/QualityObject.h (hunk 1 above)
Comment (knopers8) [reply Framework/include/QualityControl/QualityObject.h:127]: yes

**Pull request merged**